webapp/packages/plugin-administration/src/ConfigurationWizard/ServerConfiguration/ServerConfigurationPage.tsx

```diff
@@ -16,71 +16,89 @@ import {
+import { NotificationService } from '@cloudbeaver/core-events';
+import { getFirstException } from '@cloudbeaver/core-utils';
 
 import { ServerConfigurationConfigurationForm } from './Form/ServerConfigurationConfigurationForm';
 import { ServerConfigurationFeaturesForm } from './Form/ServerConfigurationFeaturesForm';
 import { ServerConfigurationInfoForm } from './Form/ServerConfigurationInfoForm';
 import { ServerConfigurationNavigatorViewForm } from './Form/ServerConfigurationNavigatorViewForm';
 import { ServerConfigurationSecurityForm } from './Form/ServerConfigurationSecurityForm';
+import { getServerConfigurationFormPart } from './getServerConfigurationFormPart';
 import { ServerConfigurationDriversForm } from './ServerConfigurationDriversForm';
+import { ServerConfigurationFormStateManager } from './ServerConfigurationFormStateManager';
 import style from './ServerConfigurationPage.module.css';
 import { ServerConfigurationService } from './ServerConfigurationService';
 
 export const ServerConfigurationPage: AdministrationItemContentComponent = observer(function ServerConfigurationPage({ configurationWizard }) {
   const translate = useTranslate();
   const styles = useS(style);
-  const [focusedRef, state] = useFocus<HTMLFormElement>({ focusFirstChild: true });
-  const service = useService(ServerConfigurationService);
-  const serverConfigResource = useService(ServerConfigResource);
-  const defaultNavigatorSettingsResource = useService(DefaultNavigatorSettingsResource);
+  const [focusedRef, ref] = useFocus<HTMLFormElement>({ focusFirstChild: true });
+  const serverConfigurationService = useService(ServerConfigurationService);
   const commonDialogService = useService(CommonDialogService);
-  const configurationWizardService = useService(ConfigurationWizardService);
-  const changed = serverConfigResource.isChanged() || defaultNavigatorSettingsResource.isChanged();
-  useFormValidator(service.validationTask, state.reference);
+  const notificationService = useService(NotificationService);
+  const serverConfigurationFormStateManager = useService(ServerConfigurationFormStateManager);
+
+  const formState = serverConfigurationFormStateManager.formState!;
+  const part = getServerConfigurationFormPart(formState);
+
+  useAutoLoad(ServerConfigurationPage, [part]);
+  useFormValidator(formState.validationTask, ref.reference);
 
   function handleChange() {
-    service.changed();
+    if (configurationWizard) {
+      serverConfigurationService.setDone(false);
+    }
 
-    if (!service.state.serverConfig.adminCredentialsSaveEnabled) {
-      service.state.serverConfig.publicCredentialsSaveEnabled = false;
+    if (!part.state.serverConfig.adminCredentialsSaveEnabled) {
+      part.state.serverConfig.publicCredentialsSaveEnabled = false;
     }
   }
 
-  function reset() {
-    service.loadConfig(true);
-  }
+  const changed = part.isChanged;
```

Comment:
currently it is a function so it is not working properly
after the merge of bad.setState ticket should be fine I suppose. but please check it after the merge wether it is working correctly

Reply:
I've tested it on your branch